src/System.Net.WebSockets/src/System/Net/WebSockets/WebSocketException.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Net.WebSockets
+{
+    public sealed class WebSocketException : Win32Exception
```

Comment:
@davidsh, is Win32Exception the right base class for this i a cross-platform world?  I'd have expected it to just derive from Exception.